### PR TITLE
Initial implementation with an example of use

### LINKED ISSUE #2: Add tests
We need to figure out (or find, haven't looked yet!) some nice abstractions on top of which we can write tests for other directives too.

### PR DESCRIPTION
**Pull request by czystyl**
Tests:
screen shot 2018-03-20 at 10 05 05

close #2

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+module.exports = {
+  extends: ['airbnb-base', 'prettier'],
```
Comment (thymikee) [inline .eslintrc.js]: add `plugin:jest/recommended` or just `jest/recommended` (if it works)

File: index.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { GraphQLID } from 'graphql';
+import { SchemaDirectiveVisitor } from 'graphql-tools';
+import { createHash } from 'crypto';
+
+class UniqueIdDirective extends SchemaDirectiveVisitor {
+  visitObject(type) {
+    // @TODO: implement directive declaratin
```
Comment (thymikee) [inline index.js]: is this todo still valid?

File: example/src/schema.graphql [hunk 3]
```diff
@@ -0,0 +1,8 @@
+type Person @uniqueID(from: ["personId", "name"]) {
```
Comment (thymikee) [inline example/src/schema.graphql]: Shouldn't this be `@uid`?

Comment (JCMais) [reply example/src/schema.graphql]: `@uid` sounds and looks better

**Comment by JCMais** [comment]
great job!

File: example/__tests__/person.test.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    persons {
+      uid
+      personId
+      name
+    }
+  }
+`;
+
+describe('Person query', () => {
+  it('Check if query return array', async () => {
```
Comment (thymikee) [inline example/__tests__/person.test.js]: nit: use lowercase for all test names
nit: return array -> returns an array

File: example/__tests__/person.test.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  it('Check array length', async () => {
+    const response = await graphql(schema, query);
+    expect(response.data.persons).toHaveLength(4);
+  });
+  it('Check recived object shape', async () => {
+    const response = await graphql(schema, query);
+
+    expect(response.data.persons[0]).toHaveProperty('uid');
+    expect(response.data.persons[0]).toHaveProperty('personId');
+    expect(response.data.persons[0]).toHaveProperty('name');
```
[the remark this answers is not in the document]
Comment (thymikee) [reply example/__tests__/person.test.js]: With this change you can remove 2 other tests

**Comment by czystyl** [comment]
Done 👍 Thanks for review

File: example/__tests__/person.test.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+      __typename
+      uid
+      personId
+      name
+    }
+  }
+`;
+
+describe('Person query', () => {
+  it('Check recived object shape', async () => {
```
Comment (AndreiCalazans) [inline example/__tests__/person.test.js]: received 😉

Comment (thymikee) [reply example/__tests__/person.test.js]: Yea, also "Check" should be lowercase. We're happy to merge PRs with such changes! :)

Comment (thymikee) [reply example/__tests__/person.test.js]: Also I'd delete the `describe` block. It's usually unnecessary complication in tests.

**Pull request merged**